### PR TITLE
Speech to text

### PR DESCRIPTION
**Pull request by VijaydityaSarker**
# PR Details

## TODOs
- [x] Have you assigned "reviewers" and "assignees"?
- [x] Have you written enough comments?
- [x] Have you tested the behaviour?

## What did you use for verifying your implementation?
- [x] Unity
- [x] HoloLens2

## Related links

- Jira:

### CODE REVIEW

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToTextButton.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToTextButton.cs]: Comments please. :)

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToTextButton.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToTextButton.cs]: This is a good start for testing functionality. I would like to see this converted to a service like our other Azure service calls before it is merged.

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToTextButton.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToTextButton.cs]: Personal preference, but would like to see 'var' replaced with strongly-typed types when not dealing with generics. So in this case the recognizer should be a SpeechRecognizer object.

**Review by aconnolly2 — CHANGES_REQUESTED** [review]
Great test of functionality. Let's convert this to a reusable service and we'll see how we can best implement this into the project's functionality.

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs]: Remove to decouple.

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs]: Should return a string type.

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs]: Remove these and just return the result as a string.

Comment (aconnolly2) [reply Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs]: Also ensure that every path of the conditional returns something (ideally the error as a string.

**Review by aconnolly2 — CHANGES_REQUESTED** [review]
Looking better. Thanks Vijay!

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs]: Nested classes could cause problems. Please move these classes outside of the main class. (They're publicly accessible anyway, so they should be fine if just moved outside of the parent class.

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs:143]: Good error handling! 👍

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs:18]: Using [SerializeField] instead of public is best practice. Good job! Proud of you :P

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs:82]: Inline comments don't need the /// format. Use /// for functions and classes (and maybe the occasional variable that needs extra explaining). Otherwise // will suffice without the <summary> tags. This doesn't need to change for my approval, but just something to keep in mind going forward. (I do prefer/// to no comments though for sure!)

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs]: This one can be removed since we're no longer tying this service to a 'button.'

**Review by aconnolly2 — CHANGES_REQUESTED** [review]
Really close. Mostly just fix the nested classes and we should be good. Everything works great in editor now. Good work Vijay!

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs
Comment (VijaydityaSarker) [reply Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/AzureSpeechToText.cs]: removed

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs
Comment (VijaydityaSarker) [reply Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs]: I have moved the nested classes outside the scope of main class SpeechToTextDisplay.It should be fine now.
Thank you for the detailed review , it helps a lot

File: Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs
Comment (aconnolly2) [inline Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs]: Please correct indentation on this and wrapper class

**Review by aconnolly2 — APPROVED** [review]
Looks good, working in editor. Hopefully all runs smoothly on the device.

**Review by satttto — APPROVED** [review]

**Review by satttto — APPROVED** [review]

**Pull request merged**
